concrete/controllers/dialog/type/update_from_type.php

```diff
@@ -0,0 +1,275 @@
+<?php
+
+namespace Concrete\Controller\Dialog\Type;
+
+use Concrete\Controller\Backend\UserInterface as BackendInterfaceController;
+use Concrete\Core\Page\PageList;
+use Concrete\Core\Foundation\Queue\Queue;
+use PageTemplate;
+use PageType;
+use Permissions;
+
+class UpdateFromType extends BackendInterfaceController
+{
+    protected $viewPath = '/dialogs/type/update_from_type';
+
+
+    private function fetchTypeAndTemplate($ptID, $pTemplateID){
+      $cmp = new \Permissions($this->pagetype);
+
+      if (!$cmp->canEditPageType()) {
+        throw new \Exception(t('You do not have access to edit this page type.'));
+      }
+
+      $this->template = PageTemplate::getByID($pTemplateID);
+
+      if (!$this->template) {
+        throw new \Exception(t('No suchpage template.'));
+      }
+    }
+
+    public function on_start()
+    {
+        parent::on_start();
+
+        $request = $this->request;
+        $ptID = $request->get('ptID');
+
+        if (!$ptID || !$request->get('ptID')) {
+          throw new \Exception(t('Invalid request data.'));
+        }
+
+        $this->pagetype = PageType::getByID($ptID);
+
+        if (!is_object($this->pagetype)) {
+            throw new \Exception('Invalid page type');
+        }
+
+        $this->permissions = new \Permissions($this->pagetype);
+    }
+
+    public function view($ptID, $pTemplateID)
+    {
+        $this->fetchTypeAndTemplate($ptID, $pTemplateID);
+
+        $pl = new PageList();
+        $pl->filterByPageTypeID($this->pagetype->getPageTypeID());
+        $pl->filterByPageTemplate($this->template);
+        $pl->ignorePermissions();
```

Comment:
This can allow people that don't have access to a page to see it. We shouldn't do this.